File: 4-minimum_of_two/task.erl
-module(task).

-export([min2/2]).

min2(Count, Values) ->
  min2(Count, Values, []).

min2(_, [], Result) ->
  lists:reverse(Result);
min2(Count, [{Value1, Value2}|ValuesTail], Result) ->
  Value3 = case Value1 < Value2 of
    true -> Value1;
    false -> Value2
  end,
  min2(Count - 1, ValuesTail, [Value3|Result]).

File: 4-minimum_of_two/task.ex
defmodule Task do

  def min2(count, values) do
    min2(count, values, [])
  end

  defp min2(_, [], result) do
    Enum.reverse(result)
  end

  defp min2(count, [{value1, value2}|tail_values], result) do
    value3 = case value1 < value2 do
      true -> value1
      false -> value2
    end
    min2(count - 1, tail_values, [value3|result])
  end

end
